source/navier_stokes/parabolic_solver.template.h

```diff
@@ -813,7 +831,27 @@ namespace ryujin
               m_i_new[d] = rho_i * get_entry<T>(velocity_.block(d), i);
             }
 
-            const auto rho_e_i_new = rho_i * get_entry<T>(internal_energy_, i);
+            auto rho_e_i_new = rho_i * get_entry<T>(internal_energy_, i);
+
+            if (crank_nicolson_extrapolation) {
+              m_i_new = Number(2.0) * m_i_new - view.momentum(U_i);
+              rho_e_i_new =
+                  Number(2.0) * rho_e_i_new - view.internal_energy(U_i);
+            }
+
+            if (!(T(0.) == std::max(T(0.), rho_i * e_min_old - rho_e_i_new))) {
+#ifdef DEBUG_OUTPUT
+              std::cout << std::fixed << std::setprecision(16);
+              const auto e_i_new = rho_e_i_new / rho_i;
+              std::cout << "Bounds violation: internal energy (critical)!\n"
+                        << "\t\te_min_old:         " << e_min_old << "\n"
+                        << "\t\te_min_old (delta): "
+                        << negative_part(e_i_new - e_min_old) << "\n"
+                        << "\t\te_min_new:         " << e_i_new << "\n"
+                        << std::endl;
+#endif
+              restart_needed = true;
+            }
```

Comment:
@ejtovar We will need to add one more workaround to make this fully robust: Either
  (a) If we have already restarted and are still in a state of bounds violation then we should simply fall back to a single backward Euler step..., and/or
  (b) combine this with an FCT limiter for the internal energy.